flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CollectionFunctionsITCase.java

```diff
@@ -479,4 +481,309 @@ private Stream<TestSetSpec> arrayUnionTestCases() {
                                 "Invalid input arguments. Expected signatures are:\n"
                                         + "ARRAY_UNION(<COMMON>, <COMMON>)"));
     }
+
+    private Stream<TestSetSpec> arrayJoinTestCases() {
+        return Stream.of(
+                TestSetSpec.forFunction(BuiltInFunctionDefinitions.ARRAY_JOIN)
+                        .onFieldsWithData(
+                                new String[] {"abv", "bbb", "cb"},
+                                new String[] {"a", "b", null},
+                                new String[] {null, "1", null},
+                                new String[] {null, null, "1", null, null},
+                                null,
+                                null,
+                                new Row[] {
+                                    Row.of(true, LocalDate.of(2022, 4, 20)),
+                                    Row.of(true, LocalDate.of(1990, 10, 14)),
+                                    null
+                                },
+                                new Integer[] {1, 2, 3, null},
+                                new Boolean[] {null, false, true},
+                                new Double[] {1.2, 34.0, 4.0, 4.5},
+                                1)
+                        .andDataTypes(
+                                DataTypes.ARRAY(DataTypes.STRING()).notNull(),
+                                DataTypes.ARRAY(DataTypes.STRING()),
+                                DataTypes.ARRAY(DataTypes.STRING()),
+                                DataTypes.ARRAY(DataTypes.STRING()),
+                                DataTypes.ARRAY(DataTypes.STRING()),
+                                DataTypes.ARRAY(DataTypes.INT()),
+                                DataTypes.ARRAY(
+                                        DataTypes.ROW(DataTypes.BOOLEAN(), DataTypes.DATE())),
+                                DataTypes.ARRAY(DataTypes.INT()),
+                                DataTypes.ARRAY(DataTypes.BOOLEAN()),
+                                DataTypes.ARRAY(DataTypes.DOUBLE()).notNull(),
+                                DataTypes.INT().notNull())
+                        .testResult(
+                                call("ARRAY_JOIN", $("f0"), null),
+                                "ARRAY_JOIN(f0, null)",
+                                null,
+                                DataTypes.STRING())
+                        .testResult(
+                                call("ARRAY_JOIN", $("f0"), "+", null),
+                                "ARRAY_JOIN(f0, '+', null)",
+                                null,
+                                DataTypes.STRING())
```

Comment:
This one is the test case of
array != null && delimiter != null && nullReplacement[] == null

Reply:
ah, thanks! missed it